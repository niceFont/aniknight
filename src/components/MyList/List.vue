<template>
  <v-row class="mt-10" justify="center">
    <v-col cols="8">
      <v-list v-if="myList.length > 0">
        <v-card class="d-flex flex-row mb-2" v-for="item in myList" :key="item.id">
          <v-row>
            <v-col cols="4" class="d-flex align-center justify-center">
              <v-avatar size="64">
                <v-img :src="item.src"></v-img>
              </v-avatar>
            </v-col>
            <v-col class="d-flex align-center justify-space-around">
              <h3>{{item.title}}</h3>
            </v-col>
            <v-col cols="2" class="d-flex align-center">
              <v-btn text :to="item.url">Visit</v-btn>
            </v-col>
            <v-col cols="2" class="d-flex align-center">
              <v-btn @click="remove(item.id)" text icon color="error">
                <v-icon>mdi-minus-circle</v-icon>
              </v-btn>
            </v-col>
          </v-row>
        </v-card>
      </v-list>
      <h3 v-if="!myList.length">No Animes found.</h3>
    </v-col>
  </v-row>
</template>

<script>
import { mapState, mapActions } from "vuex";

export default {
  name: "List",
  computed: {
    ...mapState({
      myList: state => state.myList
    })
  },
  methods: {
    remove(id) {
      this.$store.dispatch("removeAnime", { id });
    },
    ...mapActions(["removeAnime"])
  }
};
</script>

<style>
</style>