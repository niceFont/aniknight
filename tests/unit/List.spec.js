import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from "vue-router"
import Vue from 'vue'
import { mount, createLocalVue } from "@vue/test-utils"
import List from "../../src/components/MyList/List.vue"

describe("List", () => {
    Vue.use(VueRouter)
    let localVue = createLocalVue()
    let actions
    let wrapper
    let store
    let router

    beforeEach(() => {
        localVue.use(Vuetify)
        localVue.use(Vuex)

        actions = {
            removeAnime: jest.fn()
        }

        store = new Vuex.Store({
            state: {
                myList: [
                    { id: 1, title: "test", url: "testUrl", src: "link" }
                ]
            },
            actions
        })

        router = new VueRouter()

        wrapper = mount(List, {
            store,
            localVue,
            stubs: ['router-link', 'router-view'],
            router,
            created() {
                this.$vuetify.lang = {
                    t: () => { }
                }
                this.$vuetify.theme = { dark: false }

            }
        })

    })


    it("should render", () => {
        expect(wrapper.isVueInstance()).toBe(true)
    })

    it("should match snapshot", () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    it("should render data from store", () => {
        expect(wrapper.find("h3").text()).toBe("test")
    })

    it("should call action", () => {
        wrapper.find("button").trigger("click")
        expect(actions.removeAnime).toHaveBeenCalled()
    })

})