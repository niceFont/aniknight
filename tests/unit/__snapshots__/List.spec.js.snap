// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<div
  class="row mt-10 justify-center"
>
  <div
    class="col col-8"
  >
    <div
      class="v-list v-sheet v-sheet--tile theme--light"
      role="list"
    >
      <div
        class="d-flex flex-row mb-2 v-card v-sheet theme--light"
      >
        <div
          class="row"
        >
          <div
            class="d-flex align-center justify-center col col-4"
          >
            <div
              class="v-avatar"
              style="height: 64px; min-width: 64px; width: 64px;"
            >
              <div
                class="v-responsive v-image"
              >
                <div
                  class="v-image__image v-image__image--preload v-image__image--cover"
                  mode="in-out"
                  name="fade-transition"
                  style="background-position: center center;"
                />
                <div
                  class="v-responsive__content"
                />
              </div>
            </div>
          </div>
           
          <div
            class="d-flex align-center justify-space-around col"
          >
            <h3>
              test
            </h3>
          </div>
           
          <div
            class="d-flex align-center col col-2"
          >
            <router-link-stub
              class="v-btn v-btn--flat v-btn--router v-btn--text theme--light v-size--default"
            >
              <span
                class="v-btn__content"
              >
                Visit
              </span>
            </router-link-stub>
          </div>
           
          <div
            class="d-flex align-center col col-2"
          >
            <button
              class="v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default error--text"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-minus-circle theme--light"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;
